test/extended/machines/workers.go

```diff
@@ -122,64 +107,65 @@ var _ = g.Describe("[Feature:Machines][Disruptive] Managed cluster should", func
 		skipUnlessMachineAPIOperator(c.CoreV1().Namespaces())
 
 		g.By("validating node and machine invariants")
-		// fetch machines
-		machines, err := listMachines(dc, machineLabelSelectorWorker)
+		// fetch all machines
+		allMachines, err := listMachines(dc, "")
 		if err != nil {
 			e2e.Failf("unable to fetch worker machines: %v", err)
 		}
-		numMachineWorkers := len(machines)
-		if numMachineWorkers == 0 {
-			e2e.Failf("cluster should have worker machines")
+
+		if len(allMachines) == 0 {
+			e2e.Failf("cluster should have machines")
 		}
 
 		// fetch nodes
 		nodes, err := c.CoreV1().Nodes().List(metav1.ListOptions{
```

Comment:
`s/nodes/workerNodes`

Reply:
like it was obvious above that `machines` are all machines, does not same logic applies here i.e selector making it obvious that nodes are worker nodes? :)

Reply:
In the previous case `machines` and `nodes` signified worker machines and worker nodes. Which is quite obvious from the context. Here, you are listing all machines and only worker nodes. So I would recommend to go with `allMachines` and `workerNodes` since it's clear what's inside the variables. Given the `all` bit in  `allMachines` variable name does not add much context, it's fine to go with `machines`. In general, `machines` without additional prefix/suffix covers as many machines as possible based on the context. Though, it's also fine to stay with `allMachines` and `workerNodes`.